src/data_hub/dataframe.py

```diff
@@ -0,0 +1,159 @@
+        # we just made
+        if self.auto_sync:
+            self.sync(columns=[key])
+
+    def _repr_html_(self):
+        """method override to customize printing in a Jupyter notebook"""
+
+        name = self.attrs["metadata"]["name"]
+        url = construct_resource_url(
+            name=name,
+            row_type="database",
+        )
+
+        # Convert the string to a datetime object
+        date_str = self.attrs["metadata"]["dateCreated"]
+        date_obj = datetime.strptime(date_str, "%Y-%m-%d %H:%M:%S.%f")
+
+        now = datetime.now()
+
+        # Calculate the difference
+        time_diff = now - date_obj
+
+        # Convert the time difference into "x time ago" format
+        time_ago = humanize.naturaltime(time_diff)
+
+        header = f'<h4>{name} <a href = "{url}">🔗</a></h4>'
+        txt = f'<p style="font-size: 12px; color: #808080;">Created {time_ago}.</p>'
+        df_html = super()._repr_html_()
+        return header + txt + df_html
+
+    def __repr__(self):
+        """method override to customize printing in an interactive session"""
+
+        header = f'{self.attrs["metadata"]["hid"]}\n'
+        df_representation = super().__repr__()
+        return header + df_representation
+
+    def sync(
+        self,
+        *,
+        columns: Optional[list] = None,
+        rows: Optional[list] = None,
+    ):
+        """Manually synchronize data in the dataframe to the underlying Deep Origin database.
+
+        !!! tip "Deep Origin DataFrames automatically synchronize"
+            Typically, you do not need to manually synchronize. If the `auto_sync` attribute of the dataframe is set to `True`, the dataframe will automatically synchronize when changes are made to the dataframe.
+
+
+        Args:
+            columns (list, optional): The columns of the dataframe to update. Defaults to None.
+            rows (list, optional): The rows to update. Defaults to None. When None, all rows in the relevant columns are updated.
+
+        """
+
+        if columns is None:
+            columns = self.columns
+
+        for column in columns:
+            if column in ["Validation Status", "ID"]:
+                continue
+
+            column_metadata = [
+                col for col in self.attrs["metadata"]["cols"] if col["name"] == column
+            ]
+
+            if len(column_metadata) == 0:
+                raise NotImplementedError(
+                    "Column metadata not found. This is likely because it's a new column"
+                )
+
+            column_metadata = column_metadata[0]
+
+            if column_metadata["type"] == "file":
+                continue
+
+            if rows is None:
+                # we're updating a whole column
+                rows = list(self.index)
+
+            api.set_data_in_cells(
+                values=self[column],
+                row_ids=rows,
+                column_id=column,
+                database_id=self.attrs["id"],
+            )
```

Comment:
I don't know what happens behind the scenes in this call, and maybe you've already done this, but edits to 1 database can be optimized into a single HTTP request.

Reply:
this is a single request to EnsureRows